content/post/writing-a-book-remotely.md

```diff
@@ -0,0 +1,576 @@
+
+- `main` -> generate eBook releases on Leanpub
+- `preview` -> generate previews on Leanpub
+- `personal-feature-branch`
+
+Every co-author would work on their own branch and push their chapters to their dedicated branch. Once completed, we created a pull request to merge into the `preview` branch. This triggered our review process where all other co-authors reviewed the new section one after the other.
+
+What's great about this review process is that we could leverage the review features of GitHub. This allowed the reviewing person to directly make suggestions like fixing a typo or structuring the sentence. These suggestions could be then added automatically by the review owner:
+
+![Providing Feedback via GitHub Pull Requests](/img/stratospheric-ebook/github-pull-request-suggestion-example.png#center "Providing Feedback via GitHub Pull Requests")
+
+This way of working felt way more convenient than using a PDF reader, making comments and annotations to the file and then sending it over to the owner of the changes.
+
+After the review is completed by both other co-authors (six eye principle), we integrated the change into the `preview` branch to collect multiple chapters for the next release. Once every two month or so we then created the next work in progress release by merging the changes to the `main` branch.
+
+What's left to do is to click the release button inside Leanpub to build a new digital version of our eBook. Theoretically, we can even automate this and publish a new release on each commit for the `main` branch.
+
+## Getting the First Readers aka. Early Feedback 📖
+
+(having something to show, basic landing page, getting a cover, trouble with the naming, first version on Leanpub)
+
+Given the fact that we figured out how to sell the eBook and how to write it, we now had to start writing the eBook.
+
+We naturally found how to split the work and who write what based on our own interests. While Björn and I were more focussed on implementing features for the sample application and connecting various AWS services, Tom was more into the CI/CD and infrastructure setup.
+
+While we had a rough idea and an early draft for our planned table of contents, we adjusted our plans along the way.
+
+Working on a new chapter/feature usually involved some background fact checking, consulting the AWS documentation and other blogs, developing the code changes and then writing the chapter.
+
+For some concrete examples, these are chapters we worked on:
+- Getting Started with the AWS CDK
+- OAuth2 Login with AWS Cognito
+- Storing Entities to PostgreSQL using RDS
+- Structured Logging with Amazon CloudWatch
+- etc.
+
+While we had implemented similar features in the past, we still spend a good amount of time reading the AWS documentation and related blog posts to ensure our previous assumptions are correct.
+
+We also made sure every co-author works independently on their own area (like OAuth2 login or RDS setup) to avoid any misunderstanding.
```

Comment:
```suggestion
We also made sure every co-author worked independently on their own area (like OAuth2 login or RDS setup) to avoid any misunderstandings.
```